### ISSUE TITLE
research: audit JAX xp interface for user-facing introduction

### ISSUE THREAD

**Issue by Jammy2211** [Jammy2211/PyAutoArray#331]
## Overview

Audit how PyAutoArray's `xp` interface, PyAutoFit's `Analysis(use_jax=True)` flag, and the manual `jax.jit + pytree register` pattern hang together from a new user's perspective. Produce a reference design doc at `admin_jammy/notes/jax_interface.md` that the later phases of the JAX user-introduction series cite.

This is Phase 0 of a multi-phase tracker — see `PyAutoPrompt/z_features/jax_user_intro.md`. The phases that follow add `__JAX__` sections to both workspace `start_here.py` files, give `Simulator` a `use_jax=True` arg (library change), and pad out per-dataset-type JAX docs across both workspaces. This phase produces no library or workspace code change — just the design doc that the rest of the series leans on.

## Plan

- Read the xp pattern in PyAutoArray decorators; document what the user sees vs. what the contract actually is, including the "autoarray types are not JAX pytrees" rule.
- Read `Analysis(use_jax=True)` dispatch in PyAutoFit and the Analysis classes in PyAutoLens / PyAutoGalaxy. Document the path from the flag to the fitness call.
- Audit the Simulator status quo — PyAutoArray base classes have no `use_jax=True`, but `autolens_workspace/scripts/cluster/simulator.py` demonstrates the full manual pattern. Confirm and document.
- Map current workspace JAX coverage vs. target coverage (start_here + 6 dataset types × 4 script types + 4 guides × 2 workspaces).
- Form a judgement on four user-facing questions (xp leaking into user code, Simulator API, who jits, autoarray-not-pytree surfacing).
- Write the design doc that Phases 1–5 will cite. No library or workspace code changes in this phase.

<details>
<summary>Detailed implementation plan</summary>

### Affected Repositories

- **admin_jammy** (write — the output doc lands at `admin_jammy/notes/jax_interface.md`)
- PyAutoArray (read-only audit — xp interface, decorators, `Simulator` base classes)
- PyAutoFit (read-only audit — `Analysis(use_jax=True)` dispatch)
- PyAutoLens (read-only audit — `LensCalc`, `Tracer`, `Simulator` subclasses)
- PyAutoGalaxy (read-only audit — `lens_calc.py` `if xp is np` guard pattern)
- autolens_workspace (read-only audit — `scripts/cluster/simulator.py` manual jit pattern)
- autogalaxy_workspace (read-only audit — current JAX surface)

The issue lives on PyAutoArray because the xp interface is the centre of the audit and Phase 2 follow-on lands library changes here. The actual edited repo is admin_jammy.

### Work Classification

Research / design — single library-style edit on admin_jammy, no PyAutoArray code change.

### Branch Survey

| Repository | Current Branch | Dirty? |
|------------|----------------|--------|
| ./admin_jammy | main | dirty (week.md, unrelated) |
| ./PyAutoArray | main | clean |
| ./PyAutoLens | main | dirty (CLAUDE.md, README.md — ambient) |
| ./PyAutoGalaxy | main | dirty (CLAUDE.md — ambient) |
| ./autolens_workspace | main | dirty (CLAUDE.md, sample dataset files — ambient) |
| ./autogalaxy_workspace | main | dirty (CLAUDE.md, sample dataset files — ambient) |

**Suggested branch:** `feature/jax-interface-audit` (on admin_jammy only)
**Worktree root:** `~/Code/PyAutoLabs-wt/jax-interface-audit/` (created later by `/start_library`)

### Implementation Steps

1. Survey the xp pattern: read `PyAutoArray/autoarray/structures/decorators/abstract.py` end-to-end, then the three maker classes. Document what `xp` flows through and what it doesn't (e.g. autoarray wrapping is unconditional). Find every `if xp is np:` site in PyAutoArray + PyAutoGalaxy via grep.
2. Survey `Analysis(use_jax=True)`: starting from `AnalysisImaging.__init__`, trace `_use_jax` into `Fitness`, then into `figure_of_merit_from`. Document the `_vmap` vs `_jit` branch (see `PyAutoFit/autofit/non_linear/analysis/fitness.py`).
3. Survey Simulator status quo: confirm `SimulatorImaging.__init__` / `SimulatorInterferometer.__init__` accept no JAX flag; read `autolens_workspace/scripts/cluster/simulator.py` end-to-end as the working example.
4. Map workspace JAX coverage: grep for `use_jax`, `jax.jit`, `jnp`, `xp=`, `from autoconf import jax_wrapper` across both workspaces; produce a table of "current coverage vs target".
5. Form judgements on the four user-facing questions (xp leakage, Simulator API, who jits, pytree limitation).
6. Write `admin_jammy/notes/jax_interface.md` (~10–30 pages markdown). Sections: Survey, Judgement, Design, Open questions.
7. Open a PR on admin_jammy with just this file.

### Key Files (read-only audit targets)

- `PyAutoArray/autoarray/structures/decorators/abstract.py` — xp threading core
- `PyAutoArray/autoarray/dataset/imaging/simulator.py` — Simulator base (no use_jax today)
- `PyAutoFit/autofit/non_linear/analysis/fitness.py` — `_use_jax` → fitness dispatch
- `PyAutoGalaxy/autogalaxy/operate/lens_calc.py` — `if xp is np` guard pattern
- `autolens_workspace/scripts/cluster/simulator.py` — canonical manual jit example
- `autolens_workspace/start_here.py:33-50, 240-275` — current user JAX touchpoint

### Output

- `admin_jammy/notes/jax_interface.md` (new file; new `notes/` subdirectory)

</details>

## Original Prompt

<details>
<summary>Click to expand starting prompt</summary>

This issue is Phase 0 of a tracker that captures the full series. The original umbrella prompt (verbatim) is preserved in `PyAutoPrompt/z_features/jax_user_intro.md` under "__Original prompt__". The condensed Phase 0 sub-prompt that drives this issue is in `PyAutoPrompt/issued/jax_interface_audit_and_design.md`.

</details>

**Comment by Jammy2211** [comment]
Phase 0 design doc shipped: `admin_jammy/notes/jax_interface.md` (admin_jammy main, [`2f02bbf`](https://github.com/Jammy2211/admin_jammy/commit/2f02bbf)).

### Scope anchor

The user-facing design is anchored on the **core API** — `imaging`, `interferometer`, `point_source` dataset types, and the workspace guides (`data_structures`, `galaxies`, `tracer`, `lens_calc`). `cluster`, `group`, `multi`, and the developer-test paths inform the audit but are not the user surface the contract targets.

### Five recommendations

1. **`use_jax=True` is the right user-facing primitive.** `xp=np / xp=jnp` is library-author only; workspace code should never show `xp=` at the user surface.
2. **Add `Simulator.use_jax=True`** — primary deliverable for `SimulatorImaging` (imaging) and `SimulatorInterferometer` (interferometer); `PointSolver.use_jax=True` follows as a parallel for point_source. Pytree registration happens inside the simulator on first call — user never imports `register_instance_pytree` or `register_model`.
3. **User writes `@jax.jit` only at the outer call.** Inside an Analysis, the search driver does the JIT via `Fitness._vmap`. Outside (simulators, custom likelihoods), the user writes one `@jax.jit`.
4. **`use_jax=True` changes the return type contract.** Outputs come back JAX-backed. Document the host-transfer boundary; don't hide it.
5. **Surface the autoarray-not-pytree rule once, then move on.** Don't restructure autoarray types to *be* pytrees — the cost is high and the workaround is well-bounded.

### Target form for a Phase 3a `imaging/simulator.py` `__JAX Variant__`

```python
import jax
import autolens as al
import autolens.plot as aplt

simulator = al.SimulatorImaging(
    exposure_time=300.0, psf=psf, background_sky_level=0.1, use_jax=True
)

@jax.jit
def simulate(tracer):
    return simulator.via_tracer_from(tracer=tracer, grid=grid)

dataset = simulate(tracer)
aplt.fits_imaging(dataset=dataset, ...)
```

No `register_instance_pytree`. No `register_model`. No `af.Collection` mirror. No `xp=jnp` threading. One `@jax.jit`, one `use_jax=True`.

### Key findings worth flagging here

- **`use_jax=True` is the default** across every `Analysis` subclass in PyAutoLens + PyAutoGalaxy. Workspace prose should not present JAX as opt-in; it's opt-out via `Analysis(use_jax=False)` or `PYAUTO_DISABLE_JAX=1`.
- **`SimulatorImaging.via_image_from` already accepts `xp=np`**; `SimulatorInterferometer.via_image_from` does NOT. Signature inconsistency to fix in Phase 2.
- **`AnalysisImaging._register_fit_imaging_pytrees()` and `AnalysisPoint._register_fit_point_pytrees()` already auto-register** `FitImaging` / `FitPositions*` / `Tracer` / `DatasetModel` on the `use_jax=True` path. The cluster simulator's manual ceremony (~60 lines of `af.Collection` mirror + manual `register_instance_pytree(Tracer, ...)`) is the same work done by the user because cluster's simulator surface has no `use_jax=True` knob. Phase 2 closes this gap.
- **Production JAX path is `Fitness._vmap` = `vmap(jit(call))`** — not `jit(fit_from)`. Honest validation in docs should use `fitness._vmap(jnp.array(parameters))`, per the existing `autolens_workspace_test/scripts/jax_likelihood_functions/...` template.

### Next phase

Phase 1 (`workspaces/jax_start_here_intros.md`) and Phase 2 (`autoarray/simulator_use_jax.md`) are now unblocked for authoring. Per `[[feedback_no_bulk_issue_queues]]`, author each when its predecessor is close to shipping.

### Open questions flagged for discussion (full list in §4 of the doc)

- 4.1 Where does `enable_jax_for_modeling()` helper live (`al.jax.*` vs `af.jax.*`)?
- 4.2 Should `Imaging` / `Interferometer` datasets themselves be pytree-registered (so a `@jax.jit`-d simulator can return one)?
- 4.6 Should `from autoconf import jax_wrapper` warn if JAX already initialised at fp32?

Closing this issue with Phase 0 shipped.

**Comment by Jammy2211** [comment]
**Design refinement — JIT-it-yourself path** ([`f381393`](https://github.com/Jammy2211/admin_jammy/commit/f381393))

User feedback during review surfaced a real use case the original v1 covered only via the open-questions section: users wanting to wrap library methods like \`tracer.image_2d_from\` directly in \`@jax.jit\` for parameter sweeps, batch figure generation, or custom forward models (not just Simulator-shaped workloads).

After discussion, the design accepts that **the user passes \`xp=jnp\` explicitly inside their own \`@jax.jit\` body** for this path — no auto-detection from grid, no proliferation of \`use_jax=True\` onto \`Tracer\` / \`Galaxy\` / profile classes. \`@jax.jit\` + \`xp=jnp\` go together as one unit of opt-in for the advanced "JIT-it-yourself" audience; the basic Simulator / Analysis audience still sees neither.

**Recommendation 2 now bounded to three classes:** \`Simulator\`, \`PointSolver\`, \`Analysis\`. No use_jax=True proliferation.

**New §3.4.1 covers:** the \`@jax.jit\` + \`xp=jnp\` pairing rule, decorator-on-def vs \`jax.jit(bound_method)\` (both work; cache-identity footgun documented per \`[[feedback_jax_closure_cache_busts]]\`), closure-captured \`self\` vs traced-argument \`tracer\` (latter needs pytree registration, former doesn't), and the \`.array\` unwrap-rewrap pattern.

**Canonical workspace form** (Phase 5d \`lens_calc.py\` guide):

\`\`\`python
import jax, jax.numpy as jnp
import autolens as al

al.jax.enable_for_modeling()  # one-time pytree registration

@jax.jit
def image_fn(tracer, grid):
    return tracer.image_2d_from(grid=grid, xp=jnp).array

image = image_fn(tracer, grid)
image_wrapped = al.Array2D(values=image, mask=grid.mask)
\`\`\`

**New open question §4.8:** should the library raise on \`xp=np\` + jnp-backed grid mismatch? Cheap, loud, helps users discover the pairing rule the first time they break it. Recommended for Phase 2.

Issue stays closed; this is a v1.1 refinement to the shipped Phase 0 doc.
